### PR TITLE
Add iceBerg query, cohort profiling and RelationalAlgebra

### PR DESCRIPTION
**Pull request by NLGithubWP**
1. Add IceBerg query executor. aggregator, result parser.
2. In core/cohort/QueryResult, add error status.
3. In core/io/readstore, add logs
4. in core/io/readstore/FieldRS.java and CoolFieldRs, add getFieldType
5. In schema/TableSchema, add getFieldSchema and error check
6. In functionality, add CohortProfiling, IceBergLoader, RelationalAlgebra. 
7. Update ReadME
8. Add olap-tpch dataset generation scripts and files

### CODE REVIEW

File: cool-core/src/main/java/com/nus/cool/functionality/CohortProfiling.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        Float totalSum = (float) 0;
+        for (BaseResult res: results){
+            String k = res.getKey();
+            profilingCount.put(k, res.getAggregatorResult().getCount());
+            profilingSum.put(k, res.getAggregatorResult().getSum());
+            totalCount += res.getAggregatorResult().getCount();;
+            totalSum += res.getAggregatorResult().getSum();
+        }
+
+        for (String key : profilingCount.keySet()){
```
Comment (KimballCai) [inline cool-core/src/main/java/com/nus/cool/functionality/CohortProfiling.java:80]: This function is not smart enough. Please automatically decide the components of the results.

Comment (KimballCai) [reply cool-core/src/main/java/com/nus/cool/functionality/CohortProfiling.java:80]: the function should do profiling on a certain dimension or a list of dimensions

File: cool-core/src/main/java/com/nus/cool/functionality/IcebergLoader.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    /**
+     * execute query
+     *
+     * @param cube the cube that stores the data we need
+     * @param query the cohort query needed to process
+     * @return the result of the query
+     */
+
+    public static List<BaseResult> executeQuery(CubeRS cube, IcebergQuery query) throws Exception{
```
Comment (KimballCai) [inline cool-core/src/main/java/com/nus/cool/functionality/IcebergLoader.java:86]: I have moved this function into CoolModel's olapEngine. Please check.

**Review by KimballCai — COMMENTED** [review]
Please check the profiling function

File: cool-core/src/main/java/com/nus/cool/functionality/CohortProfiling.java (hunk 1 above)
Comment (NLGithubWP) [reply cool-core/src/main/java/com/nus/cool/functionality/CohortProfiling.java:80]: ok

File: cool-core/src/main/java/com/nus/cool/functionality/IcebergLoader.java (hunk 2 above)
Comment (NLGithubWP) [reply cool-core/src/main/java/com/nus/cool/functionality/IcebergLoader.java:86]: o k

**Pull request merged**